src/Kiota.Builder/CodeDOM/CodeMethod.cs

```diff
@@ -330,4 +343,20 @@ public void AddErrorMapping(string errorCode, CodeTypeBase type)
         ArgumentException.ThrowIfNullOrEmpty(errorCode);
         errorMappings.TryAdd(errorCode, type);
     }
+
+    public void AddErrorMapping(string errorCode, CodeTypeBase type, string description)
+    {
+        ArgumentNullException.ThrowIfNull(type);
+        ArgumentException.ThrowIfNullOrEmpty(errorCode);
+        errorMappings.TryAdd(errorCode, type);
+        if (!string.IsNullOrEmpty(description))
```

Comment:
should the try add result be used in the condition here?

Reply:
actually I didn't notice but we're duplicating a lot of code with the existing method.
Please:
- add the description parameter as an optional parameter to the existing method
- move the condition to the existing method
- update the condition to consider the try add, so we don't end up in a weird state where we have a description for an error type that doesn't exist
- remove this new method
- Update the GetErrorDescriptionMethod to always check whether a matching mapping is present before returning anything (same principle, safety first)